### PR TITLE
feat: add main page

### PR DESCRIPTION
**Pull request by volha1**

### CODE REVIEW

File: src/components/MovieCard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React, { ReactElement } from 'react';
+import './style.scss';
+import { Card } from '../../entity/Card';
+import menuIcon from '../../common/assets/svg/menu-icon.svg';
+
+type Movie = {
```
Comment (YuryAniskou) [inline src/components/MovieCard/index.tsx:6]: I suggest you name this type like `MovieCardProps` to make it clear whose props it is

File: src/components/MoviesList/index.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import MovieCard from '../MovieCard';
+import './style.scss';
+import { Card } from '../../entity/Card';
+
+const MoviesList = (): ReactElement => {
+  // const errorIndicator = true;
+
+  // if (errorIndicator) {
+  //   throw new Error('Server error');
+  // }
```
Comment (YuryAniskou) [inline src/components/MoviesList/index.tsx:11]: You don't need to leave commented lines in your code.

File: src/components/MoviesList/index.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  const movies: Array<Card> = [
+    new Card(1, 'Pulp Fiction', '2004', 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg', [
```
Comment (YuryAniskou) [inline src/components/MoviesList/index.tsx:14]: why do you insert a card into the array of movies? you need to use the same names. for example if it is array of movies you need to insert Movie to this array.

File: src/entity/Card.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+export class Card {
```
Comment (YuryAniskou) [inline src/entity/Card.ts:1]: you don't need to create some class for it. You can use type here and then you will not need to create some instance of this class and you will use simple object

File: src/components/ErrorBoundary/index.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import React, { ReactElement } from 'react';
+import './style.scss';
+
+type PropsType = { children: ReactElement };
+type StateType = { hasError: boolean };
+
+class ErrorBoundary extends React.Component<PropsType, StateType> {
```
Comment (YuryAniskou) [inline src/components/ErrorBoundary/index.tsx:7]: Why do you use class component here and functional component in other places?

**Comment by YuryAniskou** [comment]
I don't think that a good idea to write component code it `index.tsx` files. Usually we create `MoviesList.tsx` and in `index.ts` we export `MoviesList`

File: src/components/MovieCard/index.tsx (hunk 1 above)
Comment (volha1) [reply src/components/MovieCard/index.tsx:6]: Fixed

File: src/components/MoviesList/index.tsx (hunk 2 above)
Comment (volha1) [reply src/components/MoviesList/index.tsx:11]: This code was left to test ErrorBoundary. Anyway, I deleted it.

File: src/components/MoviesList/index.tsx (hunk 3 above)
Comment (volha1) [reply src/components/MoviesList/index.tsx:14]: Fixed.

File: src/entity/Card.ts (hunk 4 above)
Comment (volha1) [reply src/entity/Card.ts:1]: Fixed.

File: src/components/ErrorBoundary/index.tsx (hunk 5 above)
Comment (volha1) [reply src/components/ErrorBoundary/index.tsx]: Discussed with the mentor

**Pull request merged**